### PR TITLE
feat(go_utils): allow filling in cpp-specific information

### PR DESCRIPTION
**Pull request by shahms**
For C++, command line arguments aren't available to Starlark.  To work around this give `write_extra_action` some additional knowledge of `CppCompileInfo` specifically to fill out `tool` `compiler_option` and `source_file` fields.

### CODE REVIEW

File: kythe/go/util/tools/write_extra_action/write_extra_action.go [hunk 1]
```diff
@@ -80,3 +86,17 @@ func main() {
 		log.Fatal(err)
 	}
 }
+
+func populateCppCompileInfo(cc *xapb.CppCompileInfo) {
```
Comment (schroederc) [inline kythe/go/util/tools/write_extra_action/write_extra_action.go:90]: Can we add a flag to pass in this as another textproto?  Using the non-flag arguments seems too specialized for this.

Comment (shahms) [reply kythe/go/util/tools/write_extra_action/write_extra_action.go:90]: We cannot. That is the entire motivation for this change.

Comment (schroederc) [reply kythe/go/util/tools/write_extra_action/write_extra_action.go:90]: Why can't the caller structure the arguments into a textproto if it already has them?

Comment (shahms) [reply kythe/go/util/tools/write_extra_action/write_extra_action.go:90]: This only uses the trailing non-flag arguments if there is a CppCompileInfo extension provided and only after an explicit `--` delimiter, in theory allowing for some other interpretation of those arguments for a different extension.  I don't anticipate such a need as only "legacy" rules have custom protocol buffers at all and no more are likely to be added.  Extra actions being removed entirely seems more likely than new ExtraActionInfo extensions.

Comment (shahms) [reply kythe/go/util/tools/write_extra_action/write_extra_action.go:90]: The caller is Bazel. It does not provide access to the stringified command line arguments, only an opaque Args object.

**Review by schroederc — APPROVED** [review]

**Pull request merged**